### PR TITLE
create calendar page

### PR DESCRIPTION
**Pull request by SaifHayek**

### CODE REVIEW

File: client/src/components/EventCalendar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  const [initialEvents, setInitialEvents] = useState([])
+  const [interestedEvents, setInterestedEvents] = useState([])
+  const navigate = useNavigate()
+  const handleEventClick = (clickInfo:any):void => {
+    const eventId = clickInfo?.event?.extendedProps.eventId
+
+    if (window.confirm(`Are you sure you want to go to the event '${clickInfo.event.title}'`)) {
+      navigate(`/event/${eventId}`)
+    }
+  }
```
Comment (aaamra) [inline client/src/components/EventCalendar/index.tsx:20]: don't use `window.confirm` you can use a component from MUI
or you can use SweetAlert

or you don't have to confirm the user when he clicks on the event (and that's my opinion)

File: test/eventJoinedPeople.test.ts [hunk 2]
```diff
@@ -8,11 +8,12 @@ import { Message } from '../src/config/messages'
 
 beforeAll(() => build())
 afterAll(() => sequelize.close())
-
+const token = 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwidXNlcm5hbWUiOiJtb3N0YWZhIiwiaWF0IjoxNjY2NTExODQwfQ.LUdcIEgpLB_4m9p8tnQGVKwihZyqlEob_7HsQSXZUgc'
```
Comment (aaamra) [inline test/eventJoinedPeople.test.ts:11]: i thinks this token exists in our config
```suggestion
const token = config.token
```

File: src/controllers/EventParticipantsController.ts [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
+        model: Event
+      }],
+      where: { UserId: req.user?.id }
     })
     if (!allJoined) throw new CustomError(Message.NOT_FOUND, 404)
     res.json({ data: allJoined, message: Message.SUCCESS })
   }
 
   public static async store (req: Request, res:Response) {
+    console.log('yyyyy')
```
Comment (aaamra) [inline src/controllers/EventParticipantsController.ts:20]:
```suggestion
```

File: client/src/components/EventCalendar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import FullCalendar, { formatDate } from '@fullcalendar/react'
+import dayGridPlugin from '@fullcalendar/daygrid'
+import timeGridPlugin from '@fullcalendar/timegrid'
+import interactionPlugin from '@fullcalendar/interaction'
+import './style.css'
+import { useNavigate } from 'react-router-dom'
+import ApiService from '../../services/ApiService'
+
+const EventCalendar:FC = () => {
+  const [initialEvents, setInitialEvents] = useState([])
```
Comment (aaamra) [inline client/src/components/EventCalendar/index.tsx:11]:
```suggestion
  const [joindEvents, setJoindEvents] = useState([])
```

File: client/src/components/EventCalendar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  const navigate = useNavigate()
+  const handleEventClick = (clickInfo:any):void => {
+    const eventId = clickInfo?.event?.extendedProps.eventId
+
+    if (window.confirm(`Are you sure you want to go to the event '${clickInfo.event.title}'`)) {
+      navigate(`/event/${eventId}`)
+    }
+  }
+
+  const renderEventContent = (eventInfo:any):any => {
```
Comment (aaamra) [inline client/src/components/EventCalendar/index.tsx:22]: remove `any`

File: client/src/components/EventCalendar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        >
+          {eventInfo.event.title}
+
+        </p>
+        <p className={`event-status ${status}`}>{status}</p>
+      </div>
+    )
+  }
+
+  const renderSidebarEvent = (event:any):any => (
```
Comment (aaamra) [inline client/src/components/EventCalendar/index.tsx:48]: remove `any`
and you can move this to other file and make it as a react component with `props`

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by aaamra — APPROVED** [review]

**Review by shamskhodary — APPROVED** [review]

**Review by MostafaBalousha123 — APPROVED** [review]

**Review by Bakeza — APPROVED** [review]

**Pull request merged**